[lvgl] How do I use "bottom_layer," or is there an error on the SensecapDisplay?

### The problem

After switching to lvgl9, you should now use the `bottom_layer` attribute for a wallpaper. As I understand it, it should look like this:

```
lvgl:
  displays:
    - sensecap_display
  touchscreens:
    - sensecap_touchscreen
  bottom_layer:
    bg_image_src: wallpaper
  
  pages:
  ...
```
Reference Sensecap Display
```
image:
  - file: '/config/_my/_esphome/packages/lvgl_images/sensecap_wp1.jpg'
    id: wallpaper
    resize: 480x480
    type: RGB565
```
Reference Waveshare 7"
```
image:
  - file: '/config/_my/_esphome/packages/lvgl_images/7touch_wp1.jpg'
    id: wallpaper
    resize: 800x480
    type: RGB565
```


This works on a Waveshare 7" display, but on my Sensecap displays, the compilation fails with the following error.

```
Compiling .pioenvs/p-lg-sensecapbarcoderdr/src/esphome/components/output/automation.cpp.o
Compiling .pioenvs/p-lg-sensecapbarcoderdr/src/esphome/components/output/float_output.cpp.o
Compiling .pioenvs/p-lg-sensecapbarcoderdr/src/esphome/components/pca9554/pca9554.cpp.o
Compiling .pioenvs/p-lg-sensecapbarcoderdr/src/esphome/components/psram/psram.cpp.o
Compiling .pioenvs/p-lg-sensecapbarcoderdr/src/esphome/components/restart/button/restart_button.cpp.o
Compiling .pioenvs/p-lg-sensecapbarcoderdr/src/esphome/components/safe_mode/safe_mode.cpp.o
Compiling .pioenvs/p-lg-sensecapbarcoderdr/src/esphome/components/script/script.cpp.o
Compiling .pioenvs/p-lg-sensecapbarcoderdr/src/esphome/components/sensor/automation.cpp.o
In file included from src/esphome/components/lvgl/lvgl_esphome.cpp:5:
src/esphome/components/lvgl/lvgl_esphome.h: In function 'void esphome::lvgl::lv_image_set_src(lv_obj_t*, esphome::image::Image*)':
src/esphome/components/lvgl/lvgl_esphome.h:81:113: error: cannot convert 'lv_image_dsc_t*' to 'esphome::image::Image*'
   81 | inline void lv_image_set_src(lv_obj_t *obj, image::Image *image) { lv_image_set_src(obj, image->get_lv_image_dsc()); }
      |                                                                                          ~~~~~~~~~~~~~~~~~~~~~~~^~
      |                                                                                                                 |
      |                                                                                                                 lv_image_dsc_t*
src/esphome/components/lvgl/lvgl_esphome.h:81:59: note:   initializing argument 2 of 'void esphome::lvgl::lv_image_set_src(lv_obj_t*, esphome::image::Image*)'
   81 | inline void lv_image_set_src(lv_obj_t *obj, image::Image *image) { lv_image_set_src(obj, image->get_lv_image_dsc()); }
      |                                             ~~~~~~~~~~~~~~^~~~~
Compiling .pioenvs/p-lg-sensecapbarcoderdr/src/esphome/components/sensor/filter.cpp.o
Compiling .pioenvs/p-lg-sensecapbarcoderdr/src/esphome/components/sensor/sensor.cpp.o
Compiling .pioenvs/p-lg-sensecapbarcoderdr/src/esphome/components/sha256/sha256.cpp.o
*** [.pioenvs/p-lg-sensecapbarcoderdr/src/esphome/components/lvgl/lvgl_esphome.cpp.o] Error 1
========================= [FAILED] Took 11.59 seconds =========================
```

Is it possible that lvgl9 has an issue with a SEEED-INDICATOR-D1 display at 480x480px? The Waveshare 7" display has a resolution of 800x480px.
If I comment out the attribute, the compilation proceeds a bit further but crashes with dozens more errors—again, only on the SEEED-INDICATOR-D1 displays. I’ll create another issue for that.

Validate: No Errors
Clean Build Files: Done
Clean All Files: Done

### Which version of ESPHome has the issue?

2026.4.0

### What type of installation are you using?

Home Assistant Add-on

### What platform are you using?

ESP32

### Component causing the issue

lvgl

### YAML Config

```yaml

```

### Anything in the logs that might be useful for us?

```txt

```

### Additional information

_No response_

--- a/esphome/components/lvgl/lvgl_esphome.h
+++ b/esphome/components/lvgl/lvgl_esphome.h
@@ -78,15 +78,15 @@ inline void lv_style_set_text_font(lv_style_t *style, const font::Font *font) {
 #if defined(USE_LVGL_IMAGE) && defined(USE_IMAGE)
 #if LV_USE_IMAGE
 // Shortcut / overload, so that the source of an image widget can easily be updated from within a lambda.
-inline void lv_image_set_src(lv_obj_t *obj, image::Image *image) { lv_image_set_src(obj, image->get_lv_image_dsc()); }
+inline void lv_image_set_src(lv_obj_t *obj, image::Image *image) { ::lv_image_set_src(obj, image->get_lv_image_dsc()); }
 #endif  // LV_USE_IMAGE
 
 inline void lv_obj_set_style_bitmap_mask_src(lv_obj_t *obj, image::Image *image, lv_style_selector_t selector) {
-  lv_obj_set_style_bitmap_mask_src(obj, image->get_lv_image_dsc(), selector);
+  ::lv_obj_set_style_bitmap_mask_src(obj, image->get_lv_image_dsc(), selector);
 }
 
 inline void lv_obj_set_style_bg_image_src(lv_obj_t *obj, image::Image *image, lv_style_selector_t selector) {
-  lv_obj_set_style_bg_image_src(obj, image->get_lv_image_dsc(), selector);
+  ::lv_obj_set_style_bg_image_src(obj, image->get_lv_image_dsc(), selector);
 }
 #endif  // USE_LVGL_IMAGE
 #ifdef USE_LVGL_ANIMIMG

--- a/esphome/components/lvgl/lvgl_esphome.h
+++ b/esphome/components/lvgl/lvgl_esphome.h
@@ -78,14 +78,14 @@ inline void lv_style_set_text_font(lv_style_t *style, const font::Font *font) {
 #if defined(USE_LVGL_IMAGE) && defined(USE_IMAGE)
 // Shortcut / overload, so that the source of an image can easily be updated
 // from within a lambda.
-inline void lv_image_set_src(lv_obj_t *obj, image::Image *image) { lv_image_set_src(obj, image->get_lv_image_dsc()); }
+inline void lv_image_set_src(lv_obj_t *obj, image::Image *image) { ::lv_image_set_src(obj, image->get_lv_image_dsc()); }
 
 inline void lv_obj_set_style_bitmap_mask_src(lv_obj_t *obj, image::Image *image, lv_style_selector_t selector) {
-  lv_obj_set_style_bitmap_mask_src(obj, image->get_lv_image_dsc(), selector);
+  ::lv_obj_set_style_bitmap_mask_src(obj, image->get_lv_image_dsc(), selector);
 }
 
 inline void lv_obj_set_style_bg_image_src(lv_obj_t *obj, image::Image *image, lv_style_selector_t selector) {
-  lv_obj_set_style_bg_image_src(obj, image->get_lv_image_dsc(), selector);
+  ::lv_obj_set_style_bg_image_src(obj, image->get_lv_image_dsc(), selector);
 }
 #endif  // USE_LVGL_IMAGE
 #ifdef USE_LVGL_ANIMIMG

--- a/esphome/components/lvgl/lvgl_esphome.h
+++ b/esphome/components/lvgl/lvgl_esphome.h
@@ -78,15 +78,15 @@ inline void lv_style_set_text_font(lv_style_t *style, const font::Font *font) {
 #if defined(USE_LVGL_IMAGE) && defined(USE_IMAGE)
 #if LV_USE_IMAGE
 // Shortcut / overload, so that the source of an image widget can easily be updated from within a lambda.
-inline void lv_image_set_src(lv_obj_t *obj, image::Image *image) { ::lv_image_set_src(obj, image->get_lv_image_dsc()); }
+inline void lv_image_set_src(lv_obj_t *obj, image::Image *image) { lv_image_set_src(obj, image->get_lv_image_dsc()); }
 #endif  // LV_USE_IMAGE
 
 inline void lv_obj_set_style_bitmap_mask_src(lv_obj_t *obj, image::Image *image, lv_style_selector_t selector) {
-  ::lv_obj_set_style_bitmap_mask_src(obj, image->get_lv_image_dsc(), selector);
+  lv_obj_set_style_bitmap_mask_src(obj, image->get_lv_image_dsc(), selector);
 }
 
 inline void lv_obj_set_style_bg_image_src(lv_obj_t *obj, image::Image *image, lv_style_selector_t selector) {
-  ::lv_obj_set_style_bg_image_src(obj, image->get_lv_image_dsc(), selector);
+  lv_obj_set_style_bg_image_src(obj, image->get_lv_image_dsc(), selector);
 }
 #endif  // USE_LVGL_IMAGE
 #ifdef USE_LVGL_ANIMIMG

--- a/esphome/components/lvgl/lvgl_esphome.h
+++ b/esphome/components/lvgl/lvgl_esphome.h
@@ -76,9 +76,10 @@ inline void lv_style_set_text_font(lv_style_t *style, const font::Font *font) {
 }
 #endif
 #if defined(USE_LVGL_IMAGE) && defined(USE_IMAGE)
-// Shortcut / overload, so that the source of an image can easily be updated
-// from within a lambda.
+#if LV_USE_IMAGE
+// Shortcut / overload, so that the source of an image widget can easily be updated from within a lambda.
 inline void lv_image_set_src(lv_obj_t *obj, image::Image *image) { ::lv_image_set_src(obj, image->get_lv_image_dsc()); }
+#endif  // LV_USE_IMAGE
 
 inline void lv_obj_set_style_bitmap_mask_src(lv_obj_t *obj, image::Image *image, lv_style_selector_t selector) {
   ::lv_obj_set_style_bitmap_mask_src(obj, image->get_lv_image_dsc(), selector);

